r/R/type.R

```diff
@@ -109,15 +116,41 @@ Float16 <- R6Class("Float16", inherit = FixedWidthType)
 Float32 <- R6Class("Float32", inherit = FixedWidthType)
 Float64 <- R6Class("Float64", inherit = FixedWidthType)
 Boolean <- R6Class("Boolean", inherit = FixedWidthType)
-Utf8 <- R6Class("Utf8", inherit = DataType)
-LargeUtf8 <- R6Class("LargeUtf8", inherit = DataType)
-Binary <- R6Class("Binary", inherit = DataType)
-FixedSizeBinary <- R6Class("FixedSizeBinary", inherit = FixedWidthType)
-LargeBinary <- R6Class("LargeBinary", inherit = DataType)
+Utf8 <- R6Class("Utf8",
+  inherit = DataType,
+  public = list(
+    code = function() call("utf8")
+  )
+)
+LargeUtf8 <- R6Class("LargeUtf8",
+  inherit = DataType,
+  public = list(
+    code = function() call("large_utf8")
+  )
+)
+Binary <- R6Class("Binary",
+  inherit = DataType,
+  public = list(
+    code = function() call("binary")
+  )
+)
+LargeBinary <- R6Class("LargeBinary",
+  inherit = DataType, public = list(
+    code = function() call("large_binary")
+  )
+)
+FixedSizeBinary <- R6Class("FixedSizeBinary",
+  inherit = FixedWidthType,
+  public = list(
+    byte_width = function() FixedSizeBinary__byte_width(self),
+    code = function() call2("fixed_size_binary", byte_width = self$byte_width())
+  )
+)
```

Comment:
Would it be beneficial to have some sort of generic method in DataType and then an attribute in each of these classes that is the quoted string? It's not _much_ code that's repeated here, but it's a few lines per class

Reply:
Oh, actually I see below we have one for all the non-exceptionally named ones. This is probably fine since it's a limited number of classes where we need to do it

Reply:
yeah, and some require additional arguments, i.e. they're not just a `call("<string>")`, e.g. this one needs `byte_width`.